pages/docs/learn/inngest-functions.mdx

```diff
@@ -0,0 +1,104 @@
+import IconConcurrency from 'src/shared/Icons/FlowControl/Concurrency';
+
+export const description = 'Learn what Inngest functions are and of what they are capable.';
+
+# Inngest Functions
+
+Inngest functions enable developers to create and manage event-driven workflows. They offer a flexible way to build and manage event-driven workflows and background tasks, providing robust tools for retrying, scheduling, and coordinating complex sequences of operations.
+
+This page covers components of an Inngest function, as well as introduces different kinds of functions. If you'd like to learn more about Inngest's execution model, check the ["How Inngest functions are executed"](/docs/learn/how-functions-are-executed) page.
+
+## Anatomy of an Inngest function
+
+Let's have a look at the following Inngest function:
+
+```ts
+import { inngest } from "./client";
+
+export default inngest.createFunction(
+    // config
+  { id: "import-product-images" },
+    // trigger (event or cron)
+  { event: "shop/product.imported" },
+    // handler function
+    async ({ event, step }) => {
+      // here goes the business logic
+      return copyAllImagesToS3(event.data.imageURLs);
+      // if you need to perform more than one task,
+      // wrap each one in step.run()
+    };
+);
+```
+
+The above code can be translated as:
+
+> This Inngest function is called `import-product-images`. When an event called `shop/product.imported` is received, run a function called `copyAllImagesToS3`.
+
+Let's have a look at each of this function's components.
+
+<Callout>
+💡 You can test Inngest functions using standard tooling such as Jest or Mocha. To do so, export the job code and run standard unit tests.
+</Callout>
+
+### Config
+
+The first parameter of the `createFunction` method specifies Inngest function's configuration. In the above example, the `id` is specified, which will be used to identify the function in the [Inngest Dev Server](/docs/local-development):
```

Comment:
**suggestion**: We omit "Dev Server" here as this ID is used in cloud as well. Specifically, the ID is used within an _App_, but that might be too much detail. We can just leave this as "which will be used to identify the function in the Inngest system."

With the screen shot, you can maybe add: "You can see this ID in the Dev server's function list"

Reply:
Oh, right, of course.